contracts/rollup.sol

```diff
@@ -361,17 +368,28 @@ contract Rollup is RollupHelpers {
         // start with false state
         bool isDisputeValid = false;
 
+        if (batches[_batch_id - 1].batchType == Types.BatchType.Airdrop) {
+            // Validate Reddit signature and sub reddit approval
+        }
```

Comment:
Only airdrop needs batch level checks, but not monthly consent and execution. Not sure if other future features need batch level checks.